sdk/javascript/test/utils/BufferView_spec.js

```diff
@@ -0,0 +1,186 @@
+const { BufferView } = require('../../src/utils/BufferView');
+const { expect } = require('chai');
+
+describe('BufferView', () => {
+	describe('constructor', () => {
+		it('can create around whole buffer', () => {
+			// Arrange:
+			const byteArray = new Uint8Array([0, 1, 2, 3, 4, 5, 6]);
+
+			// Act:
+			const reader = new BufferView(byteArray);
+
+			// Assert:
+			expect(reader.byteArray).to.deep.equal(new Uint8Array([0, 1, 2, 3, 4, 5, 6]));
+			expect(reader.byteArray.buffer).to.equal(byteArray.buffer);
+		});
+
+		it('can create around view', () => {
+			// Arrange:
+			const byteArray = new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8]);
+			const view = new Uint8Array(byteArray.buffer, 2, 5);
+
+			// Act:
+			const reader = new BufferView(view);
+
+			// Assert:
+			expect(reader.byteArray).to.deep.equal(view);
+			expect(reader.byteArray).to.deep.equal(new Uint8Array([2, 3, 4, 5, 6]));
+
+			// - underlying buffer is the same
+			expect(reader.byteArray.buffer).to.equal(byteArray.buffer);
+			expect(reader.byteArray.byteOffset).to.equal(view.byteOffset);
+			expect(reader.byteArray.byteLength).to.equal(view.byteLength);
+		});
+	});
+
+	class TestContext {
+		constructor() {
+			this.byteArray = new Uint8Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
+			this.subView = new Uint8Array(this.byteArray.buffer, 2, 5);
+			this.view = new BufferView(this.subView);
+		}
+	}
+
+	const assertModifyTest = (mutator, expected) => {
+		// Arrange:
+		const context = new TestContext();
+
+		// Act:
+		const result = mutator(context.view);
+
+		// Assert:
+		expect(result).to.deep.equal(new Uint8Array(expected.buffer));
+
+		// - underlying buffer is the same
+		expect(result.buffer).to.equal(context.byteArray.buffer);
+		expect(result.byteOffset).to.equal(context.subView.byteOffset);
+		expect(result.byteLength).to.equal(expected.size);
+	};
+
+	describe('shrinked_buffer', () => {
+		const assertShrinkedBuffer = (newSize, expectedBuffer) => {
+			assertModifyTest(
+				reader => reader.shrinked_buffer(newSize),
+				{ buffer: expectedBuffer, size: newSize }
+			);
+		};
+
+		it('creates subview', () => {
+			assertShrinkedBuffer(3, [2, 3, 4]);
+		});
+
+		it('can create non-shrinking subview', () => {
+			assertShrinkedBuffer(5, [2, 3, 4, 5, 6]);
+		});
+
+		it('cannot create growing subview', () => {
+			// Arrange:
+			const context = new TestContext();
+
+			// Assert:
```

Comment:
`Act + Assert` when throwing (multiple)

Reply:
i think it is Act+Assert everywhere

Reply:
throw it into https://github.com/symbol/symbol-docs/blob/main/source/handbook/all-coding-guidelines.rst ?

Reply:
feel free ^^